airflow/models/baseoperator.py

```diff
@@ -1184,7 +1199,23 @@ def render_template_fields(
         """
         if not jinja_env:
             jinja_env = self.get_template_env()
-        self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
+
+        if self.template_all_fields:
+            template_fields = (
+                {key for key in self.__dict__.keys() if isinstance(getattr(self, key), str)}
+                - set(BaseOperator.__dict__.keys())
+                - set(MappedOperator.__dict__.keys())
+            )
```

Comment:
Does not looks right MappedOperator is relevant here, it breaks the OOP principles.

Reply:
For this if I only included the attributes from `BaseOperator` then some attributes get leaked through from the `MappedOperator`. This is the result of test when not removing additional attributes from `MappedOperator`.

```{'weight_rule', 'arg3', 'queue', 'trigger_rule', 'owner', 'arg1', 'arg2', 'task_id'}```

With both all the attributes belonging to the parent classes are excluded.

```{'arg1', 'arg2', 'arg3'}```